smoketests/tests/add_remove_index.py

```diff
@@ -72,7 +72,7 @@ def test_add_then_remove_index(self):
         # There are no indices, resulting in an unsupported unindexed join.
         self.publish_module(name, clear = False)
         with self.assertRaises(Exception):
-            self.subscribe(self.JOIN_QUERY, n = 0)()
+            self.subscribe(self.JOIN_QUERY, n = 0)
```

Comment:
if we've fixed the race condition, this change from #1536 is no longer necessary